File: frontend/src/history.ts
import type { AnalyzeResultV2 } from './types'
import type { IndicatorsState } from './indicators/schema'

export type HistoryItem = {
  id: string
  createdAt: number
  sourceLabel: string
  reportText: string
  result: AnalyzeResultV2
  mode?: 'form' | 'text'
  indicators?: IndicatorsState
}

File: backend/app/models/history.py
from __future__ import annotations

from typing import Any, Literal, Optional

from pydantic import BaseModel, Field


class HistoryItemIn(BaseModel):
    createdAt: int = Field(..., description="Unix ms timestamp")
    sourceLabel: str
    reportText: str
    result: dict[str, Any]
    mode: Literal["form", "text"] | None = None
    indicators: dict[str, Any] | None = None


class HistoryItemOut(HistoryItemIn):
    id: str

